test/rules-google-enhanced-ecommerce-ga4-fullstory.spec.ts

```diff
@@ -0,0 +1,616 @@
+            ecommerce: {
+              items: [
+                {
+                  item_id: 'sku_123',
+                  item_name: 'first item',
+                  price: '1.23',
+                },
+                {
+                  item_id: 'sku_456',
+                  item_name: 'second item',
+                  price: '4.56',
+                },
+              ],
+              item_list_name: 'some list',
+            },
+          });
+        });
+
+        const [eventName, eventProps] = await testHarness.popEvent();
+        expectEqual(eventName, 'select_item');
+        expectEqual(eventProps.item_id, 'sku_123');
+        expectEqual(eventProps.item_name, 'first item');
+        expectEqual(eventProps.price, 1.23);
+        expect(eventProps.item_list_name).to.be.undefined;
+      });
+
+      /*
+      Event reference:
+      https://developers.google.com/analytics/devguides/collection/ga4/reference/events?client_type=gtag#select_item
+
+      Examples:
+      https://developers.google.com/analytics/devguides/collection/ga4/ecommerce?client_type=gtag#select_an_item_from_a_list
+      */
+      it('reads select_item gtag event', async () => {
+        await testHarness.execute(() => {
+          globalThis.dataLayer.push([
+            'event',
+            'select_item',
+            {
+              items: [
+                {
+                  item_id: 'sku_123',
+                  item_name: 'first item',
+                  price: '1.23',
+                },
+                {
+                  item_id: 'sku_456',
+                  item_name: 'second item',
+                  price: '4.56',
+                },
+              ],
+              item_list_name: 'some list',
+            },
+          ]);
+        });
+
+        const [eventName, eventProps] = await testHarness.popEvent();
+        expectEqual(eventName, 'select_item');
+        expectEqual(eventProps.item_id, 'sku_123');
+        expectEqual(eventProps.item_name, 'first item');
+        expectEqual(eventProps.price, 1.23);
+        expect(eventProps.item_list_name).to.be.undefined;
+      });
+
+      /*
+      Event reference:
+      https://developers.google.com/analytics/devguides/collection/ga4/reference/events?client_type=gtm#view_item
+
+      Examples:
+      https://developers.google.com/analytics/devguides/collection/ga4/ecommerce?client_type=gtm#view_item_details
+      */
+      it('reads view_item gtm event', async () => {
+        await testHarness.execute(() => {
+          globalThis.dataLayer.push({
+            event: 'view_item',
+            ecommerce: {
+              items: [
+                {
+                  item_id: 'sku_123',
+                  item_name: 'first item',
+                  price: '1.23',
+                },
+                {
+                  item_id: 'sku_456',
+                  item_name: 'second item',
+                  price: '4.56',
+                },
+              ],
+              currency: 'USD',
```

Comment:
Would we want to capture root level properties like `currency`?

Reply:
Capturing the first `items` element and root level properties would require a new operator which is out of scope but could be considered if we get feedback that root level properties are valuable.